core/blockchain.go

```diff
@@ -183,11 +183,12 @@ type BlockChain struct {
 	running       int32          // 0 if chain is running, 1 when stopped
 	procInterrupt int32          // interrupt signaler for block processing
 
-	engine     consensus.Engine
-	validator  Validator  // Block and state validator interface
-	prefetcher Prefetcher // Block state prefetcher interface
-	processor  Processor  // Block transaction processor interface
-	vmConfig   vm.Config
+	engine         consensus.Engine
+	validator      Validator             // Block and state validator interface
+	triePrefetcher *state.TriePrefetcher // Trie prefetcher interface
```

Comment:
Suddenly have an idea, does it makes sense to put the notion of `triePrefetcher` into the state?
Whenever the state is created, then `triePrefetcher ` is also created if `snapshot` is enabled.
If we throw away the state, the `triePrefetcher` is deallocated silently.